lib/rucio/daemons/hermes/hermes.py

```diff
@@ -517,6 +517,36 @@ def aggregate_to_influx(
     return 204
 
 
+def build_message_dict(
+        bulk: int,
+        worker_number: int,
+        total_workers: int,
+        message_dict: dict[str, dict[str, Any]],
+        logger: "LoggerFunction",
+        service: str = ""
+) -> None:
+    start_time = time.time()
+    messages = retrieve_messages(
+        bulk=bulk,
+        old_mode=False,
+        thread=worker_number,
+        total_threads=total_workers,
+        service_filter=service
+    )
+
+    if messages:
+        if service not in message_dict:
+            message_dict[service] = []
+            for message in messages:
+                message_dict[service].append(message)
+            logger(
+                logging.DEBUG,
+                "Retrieved %i messages retrieved in %s seconds",
+                len(messages),
+                time.time() - start_time,
+            )
```

Comment:
This could be simplified to:

```suggestion
    if messages and service not in message_dict:
        message_dict[service] = messages.copy()
        logger(
            logging.DEBUG,
            "Retrieved %i messages retrieved in %s seconds",
            len(messages),
            time.time() - start_time,
        )
```

(I'm not sure if `messages` needs to be copied here, but it doesn't hurt. This way at least the functionality remains the same)